src/main/java/org/openstreetmap/atlas/generator/tools/caching/HadoopAtlasFileCache.java

```diff
@@ -27,21 +29,25 @@
     private static final Logger logger = LoggerFactory.getLogger(HadoopAtlasFileCache.class);
 
     private final String parentAtlasPath;
+    private final SlippyTilePersistenceScheme atlasScheme;
 
     /**
      * Create a new cache.
      *
      * @param parentAtlasPath
      *            The parent path to the atlas files. This might look like hdfs://some/path/to/files
+     * @param atlasScheme
+     *            The scheme used to locate atlas files based on slippy tiles
      * @param configuration
      *            The configuration map
      */
     public HadoopAtlasFileCache(final String parentAtlasPath,
-            final Map<String, String> configuration)
+            final SlippyTilePersistenceScheme atlasScheme, final Map<String, String> configuration)
     {
         super(new SystemTemporaryFileCachingStrategy(),
                 uri -> FileSystemHelper.resource(uri.toString(), configuration));
         this.parentAtlasPath = parentAtlasPath;
+        this.atlasScheme = atlasScheme;
```

Comment:
What happens if the user passes a `null` scheme, or does not want to pass in a scheme? It is possible downstream users may not anticipate this constructor change, or may not necessarily need this functionality. Perhaps we should overload the constructor and have a sensible default scheme?

I am planning on adding additional constructors to support the new `NamespaceCachingStrategy`, so this is something I may be able to do in a later PR.

Reply:
Yes, we can make it use defaults... Let me do that

Reply:
Done..